beacon-chain/rpc/eth/validator/handlers.go

```diff
@@ -52,61 +54,133 @@ func (s *Server) GetAggregateAttestation(w http.ResponseWriter, r *http.Request)
+func (s *Server) GetAggregateAttestationV2(w http.ResponseWriter, r *http.Request) {
+	_, span := trace.StartSpan(r.Context(), "validator.GetAggregateAttestationV2")
+	defer span.End()
+
+	_, attDataRoot, ok := shared.HexFromQuery(w, r, "attestation_data_root", fieldparams.RootLength, true)
+	if !ok {
+		return
+	}
+	_, slot, ok := shared.UintFromQuery(w, r, "slot", true)
+	if !ok {
+		return
+	}
+	_, index, ok := shared.UintFromQuery(w, r, "committee_index", true)
+	if !ok {
+		return
+	}
+	i := strconv.FormatUint(index, 10)
+	match := s.aggregateAttestation(w, primitives.Slot(slot), i, attDataRoot)
+	if match == nil {
+		return
+	}
+	resp := &structs.AggregateAttestationResponse{
+		Version: version.String(match.Version()),
+	}
+	if match.Version() >= version.Electra {
+		attPostElectra, ok := match.(*ethpbalpha.AttestationElectra)
+		if !ok {
+			httputil.HandleError(w, "Match is not of type AttestationElectra", http.StatusInternalServerError)
+			return
+		}
+		att := structs.AttElectraFromConsensus(attPostElectra)
+		data, err := json.Marshal(att)
+		if err != nil {
+			httputil.HandleError(w, "Could not marshal attestation: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+		resp.Data = data
+	} else {
+		attPreElectra, ok := match.(*ethpbalpha.Attestation)
+		if !ok {
+			httputil.HandleError(w, "Match is not of type Attestation", http.StatusInternalServerError)
+			return
+		}
+		att := structs.AttFromConsensus(attPreElectra)
+		data, err := json.Marshal(att)
+		if err != nil {
+			httputil.HandleError(w, "Could not marshal attestation: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+		resp.Data = data
+	}
+	httputil.WriteJson(w, resp)
+}
+
+func (s *Server) aggregateAttestation(w http.ResponseWriter, slot primitives.Slot, index string, attDataRoot []byte) ethpbalpha.Att {
 	var match ethpbalpha.Att
 	var err error
 
-	match, err = matchingAtt(s.AttestationsPool.AggregatedAttestations(), primitives.Slot(slot), attDataRoot)
+	match, err = matchingAtt(s.AttestationsPool.AggregatedAttestations(), slot, attDataRoot, index)
 	if err != nil {
 		httputil.HandleError(w, "Could not get matching attestation: "+err.Error(), http.StatusInternalServerError)
-		return
+		return nil
 	}
 	if match == nil {
 		atts, err := s.AttestationsPool.UnaggregatedAttestations()
 		if err != nil {
 			httputil.HandleError(w, "Could not get unaggregated attestations: "+err.Error(), http.StatusInternalServerError)
-			return
+			return nil
 		}
-		match, err = matchingAtt(atts, primitives.Slot(slot), attDataRoot)
+		match, err = matchingAtt(atts, slot, attDataRoot, index)
 		if err != nil {
 			httputil.HandleError(w, "Could not get matching attestation: "+err.Error(), http.StatusInternalServerError)
-			return
+			return nil
+		}
+		if match == nil {
+			httputil.HandleError(w, "No matching attestation found", http.StatusNotFound)
+			return nil
+		}
+		_, err = attestations.Aggregate([]ethpbalpha.Att{match})
```

Comment:
```suggestion
		agg, err = attestations.Aggregate([]ethpbalpha.Att{matches})
		...
		return agg[0]
```

If they cannot be aggregate to just one (`len(agg) == 1`), then return best by committee bits count.